lib/pyfrc/physics/drivetrains.py

```diff
@@ -140,6 +140,64 @@ def mecanum_drivetrain(lr_motor, rr_motor, lf_motor, rf_motor, x_wheelbase=2, y_
     
     return Vx, Vy, Vw
     
+def four_motor_swerve_drivetrain(lr_motor, rr_motor, lf_motor, rf_motor, lr_angle, rr_angle, lf_angle, rf_angle, x_wheelbase=2, y_wheelbase=2, speed=5):
+    '''
+        Four motors that can be rotated in any direction
+        
+        If any motors are inverted, then you will need to multiply that motor's
+        value by -1.
+        
+        :param lr_motor:   Left rear motor value (-1 to 1); 1 is forward
+        :param rr_motor:   Right rear motor value (-1 to 1); 1 is forward
+        :param lf_motor:   Left front motor value (-1 to 1); 1 is forward
+        :param rf_motor:   Right front motor value (-1 to 1); 1 is forward
+        :param lr_angle:   Left rear motor angle in degrees (0 to 360)
+        :param rr_angle:   Right rear motor angle in degrees (0 to 360)
+        :param lf_angle:   Left front motor angle in degrees (0 to 360)
+        :param rf_angle:   Right front motor angle in degrees (0 to 360)
+        :param x_wheelbase: The distance in feet between right and left wheels.
+        :param y_wheelbase: The distance in feet between forward and rear wheels.
+        :param speed:      Speed of robot in feet per second (see above)
+        
+        :returns: Speed of robot in x (ft/s), Speed of robot in y (ft/s), 
+                  clockwise rotation of robot (radians/s)
+    '''
+    # Calculate speed of each wheel
+    lr = lr_motor * speed
+    rr = rr_motor * speed
+    lf = lf_motor * speed
+    rf = rf_motor * speed
+
+    # Calculate angle in radians
+    lr_rad = lr_angle * (math.pi / 180)
+    rr_rad = rr_angle * (math.pi / 180)
+    lf_rad = lf_angle * (math.pi / 180)
+    rf_rad = rf_angle * (math.pi / 180)
+
+    # Calculate wheelbase radius
+    wheelbase_radius = math.sqrt((x_wheelbase / 2) ** 2 + (y_wheelbase/2) ** 2)
```

Comment:
Well, if you removed those parens, yes, it would definitely not do what you wanted. I would assume you remember order of operations from school (I believe the common mnemonic in the US is PEMDAS?). 😜